packages/next-intl/src/middleware/NextIntlMiddlewareConfig.tsx

```diff
@@ -37,6 +40,7 @@ export type MiddlewareConfigWithDefaults = MiddlewareConfig & {
   alternateLinks: boolean;
   localePrefix: LocalePrefix;
   localeDetection: boolean;
+  basePath: string;
```

Comment:
I did a first test with `example-next-13` in the repo. With `localePrefix: 'always'` this already seems to work quite well! Pretty cool that `Link` automatically considers the `basePath` if it's set in `next.config.js` ([source](https://nextjs.org/docs/pages/api-reference/next-config-js/basePath#links)).

With `localePrefix: 'as-necessary'` I run into infinite redirects though when trying to access the app root. Can you reproduce that?

Reply:
I didn't test locally with `localePrefix: 'as-necessary'` as we require the locale to be in the path. I'll scaffold a new next app and reproduce it.